pkg/resources/grant_privileges_to_account_role.go

```diff
@@ -0,0 +1,1103 @@
+						Summary:  "Failed to grant all privileges",
+						Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err.Error()),
+					},
+				}
+			}
+		}
+
+		id.AllPrivileges = allPrivileges.(bool)
+	}
+
+	if d.HasChange("always_apply") {
+		id.AlwaysApply = d.Get("always_apply").(bool)
+	}
+
+	if id.AlwaysApply {
+		logging.DebugLogger.Printf("[DEBUG] Performing always_apply re-grant")
+		err := client.Grants.GrantPrivilegesToAccountRole(
+			ctx,
+			getAccountRolePrivilegesFromSchema(d),
+			getAccountRoleGrantOn(d),
+			id.RoleName,
+			&sdk.GrantPrivilegesToAccountRoleOptions{
+				WithGrantOption: &id.WithGrantOption,
+			},
+		)
+		if err != nil {
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity: diag.Error,
+					Summary:  "Always apply. An error occurred when granting privileges to account role",
+					Detail:   fmt.Sprintf("Id: %s\nAccount role name: %s\nError: %s", d.Id(), id.RoleName, err.Error()),
+				},
+			}
+		}
+	}
+
+	logging.DebugLogger.Printf("[DEBUG] Setting identifier to %s", id.String())
+	d.SetId(id.String())
+
+	return ReadGrantPrivilegesToAccountRole(ctx, d, meta)
+}
+
+func DeleteGrantPrivilegesToAccountRole(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	logging.DebugLogger.Printf("[DEBUG] Entering delete grant privileges to account role")
+	db := meta.(*sql.DB)
+
+	logging.DebugLogger.Printf("[DEBUG] Creating new client from db")
+	client := sdk.NewClientFromDB(db)
+
+	id, err := ParseGrantPrivilegesToAccountRoleId(d.Id())
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to parse internal identifier",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err.Error()),
+			},
+		}
+	}
+	logging.DebugLogger.Printf("[DEBUG] Parsed identifier: %s", id.String())
+
+	err = client.Grants.RevokePrivilegesFromAccountRole(
+		ctx,
+		getAccountRolePrivilegesFromSchema(d),
+		getAccountRoleGrantOn(d),
+		id.RoleName,
+		&sdk.RevokePrivilegesFromAccountRoleOptions{},
+	)
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "An error occurred when revoking privileges from account role",
+				Detail:   fmt.Sprintf("Id: %s\nAccount role name: %s\nError: %s", d.Id(), id.RoleName, err.Error()),
+			},
+		}
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func ReadGrantPrivilegesToAccountRole(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	logging.DebugLogger.Printf("[DEBUG] Entering read grant privileges to role")
+	id, err := ParseGrantPrivilegesToAccountRoleId(d.Id())
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to parse internal identifier",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err.Error()),
+			},
+		}
+	}
+	logging.DebugLogger.Printf("[DEBUG] Parsed identifier: %s", id.String())
+
+	if id.AlwaysApply {
+		triggerId, err := uuid.GenerateUUID()
```

Comment:
@sfc-gh-jcieslak I suggested it for the previous change and I agree with Scott, that this should be documented in code as a comment because the motivation is unclear.